### PR TITLE
Simple environment to move robot fwd by 0.25m; +warning in goto_controller if goal is sent w/out robot being "active"

### PR DESCRIPTION
**Pull request by zephirefaith**
## Motivation and Context

A simple environment as an example to let user's navigate the robot forward by 0.25m.

## Changelog

- New env: SimpleNavEnv
- Warning in goto_controller node

## How Has This Been Tested

On the real robot v2

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/home_robot_hw/home_robot_hw/env/simple_navigation_env.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        self.forward_step = forward_step  # in meters
+        self.reset()
+
+    def reset(self):
+        self._current_pose = self.get_base_pose()
+
+    def apply_action(self):
+        continuous_action = np.zeros(3)
+        continuous_action[0] = 0.25
+        if continuous_action is not None:
```
Comment (exhaustin) [inline src/home_robot_hw/home_robot_hw/env/simple_navigation_env.py:26]: Why are you checking if it's None if you just assigned values to it?

File: src/home_robot_hw/home_robot_hw/env/simple_navigation_env.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
+from typing import Any, Dict, Optional
+
+import numpy as np
+import rospy
+
+import home_robot
+from home_robot_hw.env.stretch_abstract_env import StretchEnv
+
+class StretchSimpleNavEnv(StretchEnv):
```
[the remark this answers is not in the document]
Comment (zephirefaith) [reply src/home_robot_hw/home_robot_hw/env/simple_navigation_env.py:9]: Ah! I see now. This makes sense.

File: src/home_robot_hw/home_robot_hw/env/simple_navigation_env.py (hunk 1 above)
Comment (zephirefaith) [reply src/home_robot_hw/home_robot_hw/env/simple_navigation_env.py:26]: Leftover from old code which was receiving an input. Will remove this.

File: src/home_robot_hw/home_robot_hw/nodes/goto_controller.py [hunk 3]
```diff
@@ -119,12 +119,16 @@ def _run_control_loop(self):
...
+            if self.xyt_goal is not None:
+                if self.active:
+                    # Compute control
+                    v_cmd, w_cmd = self.controller.compute_control()
+
+                    # Command robot
+                    self._set_velocity(v_cmd, w_cmd)
+                else:
+                    log.warn("Received a goal while NOT active. Goal will be unset.")
+                    self.xyt_goal = None
```
[the remark this answers is not in the document]
Comment (exhaustin) [reply src/home_robot_hw/home_robot_hw/nodes/goto_controller.py:131]: Let me know if you would like me to make this change actually.

File: src/home_robot_hw/home_robot_hw/env/simple_navigation_env.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (zephirefaith) [reply src/home_robot_hw/home_robot_hw/env/simple_navigation_env.py:9]: Python actually won't let me do what you are suggesting. Since StretchEnv is an abstract-class it can not be instantiated as a variable.

File: src/home_robot_hw/home_robot_hw/nodes/goto_controller.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (zephirefaith) [reply src/home_robot_hw/home_robot_hw/nodes/goto_controller.py]: Just addressed this. Take a look?

File: src/home_robot_hw/home_robot_hw/env/simple_navigation_env.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (cpaxton) [reply src/home_robot_hw/home_robot_hw/env/simple_navigation_env.py:9]: I think we just need to refactor the StretchEnv sooner rather than later @exhaustin

File: src/home_robot_hw/home_robot_hw/env/simple_navigation_env.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    ):
+        super().__init__(*args, **kwargs)
+
+        self.forward_step = forward_step  # in meters
+        self.reset()
+
+    def reset(self):
+        self._current_pose = self.get_base_pose()
+
+    def apply_action(self):
```
Comment (cpaxton) [inline src/home_robot_hw/home_robot_hw/env/simple_navigation_env.py:23]: please add comments and types

File: src/home_robot_hw/home_robot_hw/nodes/goto_controller.py [hunk 5]
```diff
@@ -85,17 +85,18 @@ def _goal_update_callback(self, msg: Pose):
         self.goal_visualizer(pose_goal.matrix())
 
     def _enable_service(self, request):
```
Comment (cpaxton) [inline src/home_robot_hw/home_robot_hw/nodes/goto_controller.py:87]: type

File: src/home_robot_hw/home_robot_hw/nodes/goto_controller.py [hunk 6]
```diff
@@ -85,17 +85,18 @@ def _goal_update_callback(self, msg: Pose):
...
+        self.xyt_goal = None
         self.active = True
         return TriggerResponse(
```
Comment (cpaxton) [inline src/home_robot_hw/home_robot_hw/nodes/goto_controller.py:90]: this is the return type of this function - TriggerResponse - we should get in the habit of fixing these when we modify functions

File: src/home_robot_hw/home_robot_hw/nodes/goto_controller.py [hunk 7]
```diff
@@ -85,17 +85,18 @@ def _goal_update_callback(self, msg: Pose):
...
             success=True,
-            message=f"Goto controller is now RUNNING",
+            message="Goto controller is now RUNNING",
```
Comment (cpaxton) [inline src/home_robot_hw/home_robot_hw/nodes/goto_controller.py:92]: why this change?

File: src/home_robot_hw/home_robot_hw/nodes/goto_controller.py [hunk 8]
```diff
@@ -85,17 +85,18 @@ def _goal_update_callback(self, msg: Pose):
...
         )
 
     def _disable_service(self, request):
         self.active = False
         return TriggerResponse(
             success=True,
-            message=f"Goto controller is now STOPPED",
+            message="Goto controller is now STOPPED",
```
Comment (cpaxton) [inline src/home_robot_hw/home_robot_hw/nodes/goto_controller.py:99]: same comments as above

File: src/home_robot_hw/home_robot_hw/nodes/goto_controller.py [hunk 9]
```diff
@@ -119,12 +120,15 @@ def _run_control_loop(self):
...
-                self._set_velocity(v_cmd, w_cmd)
+            if self.xyt_goal is not None:
+                if self.active:
+                    # Compute control
+                    v_cmd, w_cmd = self.controller.compute_control()
+
+                    # Command robot
+                    self._set_velocity(v_cmd, w_cmd)
+                else:
+                    log.warn("Received a goal while NOT active. Goal will be unset.")
```
Comment (cpaxton) [inline src/home_robot_hw/home_robot_hw/nodes/goto_controller.py:131]: why not unset the goal here? self.xyt_goal = None

**Review by cpaxton — COMMENTED** [review]
minor stuff

File: src/home_robot_hw/home_robot_hw/nodes/goto_controller.py (hunk 7 above)
Comment (zephirefaith) [reply src/home_robot_hw/home_robot_hw/nodes/goto_controller.py:92]: Incorrect usage of f-strings (without a placeholder)

File: src/home_robot_hw/home_robot_hw/nodes/goto_controller.py (hunk 9 above)
Comment (zephirefaith) [reply src/home_robot_hw/home_robot_hw/nodes/goto_controller.py:131]: Austin suggested that unsetting goal in _enable_service ensures controller always starts in neutral state, which makes sense. That said, I think we should unset goal here as well so it doesn't keep triggering this condition. I'll add this tomorrow.

File: src/home_robot_hw/home_robot_hw/nodes/goto_controller.py (hunk 8 above)
Comment (zephirefaith) [reply src/home_robot_hw/home_robot_hw/nodes/goto_controller.py:99]: Incorrect usage of f-string.

File: src/home_robot_hw/home_robot_hw/env/simple_navigation_env.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (exhaustin) [reply src/home_robot_hw/home_robot_hw/env/simple_navigation_env.py:9]: Yeah I agree. P0 for me right now.

**Comment by zephirefaith** [comment]
@exhaustin @cpaxton Addressed your comments, take a look and if things look good then let's merge this.

**Review by exhaustin — APPROVED** [review]
Looks great! 

I was troubled by the non-straightforward conditionals in the main step method in the controller, and it just came to me that we can just do the check while setting the goal instead of checking if the goal has been incorrectly set while inactive.
I pushed directly to this branch. Can you take a look and see if it makes sense? Happy to revert if it doesn't. Thanks!

**Review by cpaxton — APPROVED** [review]
lgtm
